ibc/host_introspection_test.go

```diff
@@ -0,0 +1,271 @@
+	require.Equal(t, uint64(0), height.GetRevisionHeight())
+
+	// increment the height
+	publishNewHeightEvent(t, ibcMod.GetBus(), 1)
+
+	height, err = cm.GetCurrentHeight()
+	require.NoError(t, err)
+	require.Equal(t, uint64(1), height.GetRevisionNumber())
+	require.Equal(t, uint64(1), height.GetRevisionHeight())
+}
+
+func TestHost_GetHostConsensusState(t *testing.T) {
+	_, _, _, _, ibcMod := prepareEnvironment(t, 1, 0, 0, 0)
+	cm := ibcMod.GetBus().GetClientManager()
+
+	consState, err := cm.GetHostConsensusState(&client_types.Height{RevisionNumber: 1, RevisionHeight: 0})
+	require.NoError(t, err)
+
+	require.Equal(t, "08-wasm", consState.ClientType())
+	require.NoError(t, consState.ValidateBasic())
+	require.Less(t, consState.GetTimestamp(), uint64(time.Now().UnixNano()))
+
+	pocketConState := new(light_client_types.PocketConsensusState)
+	err = codec.GetCodec().Unmarshal(consState.GetData(), pocketConState)
+	require.NoError(t, err)
+
+	blockstore := ibcMod.GetBus().GetPersistenceModule().GetBlockStore()
+	block, err := blockstore.GetBlock(0)
+	require.NoError(t, err)
+
+	require.Equal(t, block.BlockHeader.Timestamp, pocketConState.Timestamp)
+	require.Equal(t, block.BlockHeader.StateHash, pocketConState.StateHash)
+	require.Equal(t, block.BlockHeader.StateTreeHashes, pocketConState.StateTreeHashes)
+	require.Equal(t, block.BlockHeader.NextValSetHash, pocketConState.NextValSetHash)
+}
+
+func TestHost_GetHostClientState(t *testing.T) {
+	_, _, _, _, ibcMod := prepareEnvironment(t, 1, 0, 0, 0)
+	cm := ibcMod.GetBus().GetClientManager()
+
+	clientState, err := cm.GetHostClientState(&client_types.Height{RevisionNumber: 1, RevisionHeight: 0})
+	require.NoError(t, err)
+	require.Equal(t, "08-wasm", clientState.ClientType())
+
+	pocketClientState := new(light_client_types.PocketClientState)
+	err = codec.GetCodec().Unmarshal(clientState.GetData(), pocketClientState)
+	require.NoError(t, err)
+
+	blockstore := ibcMod.GetBus().GetPersistenceModule().GetBlockStore()
+	block, err := blockstore.GetBlock(0)
+	require.NoError(t, err)
+
+	require.Equal(t, pocketClientState.NetworkId, block.BlockHeader.NetworkId)
+	require.Equal(t, pocketClientState.TrustLevel, &light_client_types.Fraction{Numerator: 2, Denominator: 3})
+	require.Equal(t, pocketClientState.TrustingPeriod.AsDuration().Nanoseconds(), int64(1814400000000000))
+	require.Equal(t, pocketClientState.UnbondingPeriod.AsDuration().Nanoseconds(), int64(1814400000000000))
+	require.Equal(t, pocketClientState.MaxClockDrift.AsDuration().Nanoseconds(), int64(900000000000))
+	require.Equal(t, pocketClientState.LatestHeight, &client_types.Height{RevisionNumber: 1, RevisionHeight: 0})
+	require.True(t, pocketClientState.ProofSpec.ConvertToIcs23ProofSpec().SpecEquals(ics23.SmtSpec))
+}
+
+func TestHost_VerifyHostClientState(t *testing.T) {
+	_, _, _, persistenceMod, ibcMod := prepareEnvironment(t, 1, 0, 0, 0)
+	cm := ibcMod.GetBus().GetClientManager()
+
+	approxTime := time.Minute * 15
```

Comment:
Lots of numbers here (and above) that would be easier to understand as constants w/ comments.